collector/filesystem_linux.go

```diff
@@ -76,6 +113,26 @@ func (c *filesystemCollector) GetStats() ([]filesystemStats, error) {
 	return stats, nil
 }
 
+// watcher listens on the given success channel and if the channel closes
+// then the watcher does nothing. If instead the timeout is reached, the
+// mount point that is being watched is marked as stuck.
+func watcher(mountPoint string, success chan struct{}, lock chan struct{}) {
+	select {
+	case <-success:
+		// Success
+	case <-time.After(mountTimeout):
+		// Timed out, mark mount as stuck
+		mutex.Lock()
+		select {
+		case <-lock:
```

Comment:
I'm not sure we need this second `lock` channel. Since we are already holding the mutex here, can't we just check again under the mutex that the `success` channel hasn't been closed yet?

Reply:
(Though I think that requires moving the closing of the `success` channel into the mutex-protected section in `GetStats()` too)

Reply:
I think you're right, for some reason I think I convinced myself of a case where it wouldn't work like that but I don't remember what it was and I don't see a way it wouldn't work now.